src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -368,10 +368,10 @@ public World getWorld()
     }
 
     @Override
-    public float getDistanceSquared(@NotNull BlockPos pos)
+    public long getDistanceSquared(@NotNull BlockPos pos)
     {
         //  Perform a 2D distance calculation, so pass center.posY as the Y
-        return BlockPosUtil.getDistanceSquared(center, new BlockPos(pos.getX(), center.getY(), pos.getZ()));
+        return BlockPosUtil.getDistanceSquared2D(center, pos);
```

Comment:
Why do we still use 2D distance for the general distance? Would it not make more sense to change all uses of this to 2D and make this 3d?